### ISSUE TITLE
Stubs rendering default props

### ISSUE THREAD

**Issue by robokozo** [vuejs/test-utils#945]
When rendering snapshots, stubbed components are actually being evaluated in order to apply the default props to the element.

For example if my component imports and contains the following code:

```      
<Button
      ref="cancel-button"
      label="Cancel"
      :disabled="isLoading"
      icon="pi pi-times"
      @click="cancelClickHandler"
      class="p-button-text"
    />
```

And I stub out the `Button` component in my test

```
mountOptions = {
      global: {
        stubs: ["Button"]
      }
    }
```
My snapshot will render with
`
<button-stub label="Cancel" icon="pi pi-times" iconpos="left" loading="false" loadingicon="pi pi-spinner pi-spin" disabled="false" class="p-button-text"></button-stub>
`

`iconpos="left"` and `loadingicon="pi pi-spinner pi-spin" ` should not be rendered.

This is annoying for me because whenever I update my `Button` component my snapshots for anything using the `Button` component need to be updated. I already have my `Button` component covered with its own tests so this is unnecessary.

**Comment by lmiller1990** [comment]
Where are `loadingicon` and `iconpos` coming from?

Snapshots + stubs are difficult to handle - seems like there are varying opinions on what should and should not be rendered. I wonder if we can provide a few different snapshot serializers, so people can choose the snapshot behavior they'd like. 🤔

**Comment by robokozo** [comment]
They are defined on the actual "Button" component that I'm stubbing out. These specific props are defined with default values that are being rendered by the snapshot. (In this particular case they are defined in the PrimeVue component library I'm using)

I'm assuming that the actual Button component is being evaluated before the stubbing process takes place.

**Comment by robokozo** [comment]
> I'm assuming that the actual Button component is being evaluated before the stubbing process takes place.

I ran into another issue which leads me to believe this is the case and the stubbing process should be examined. In this new situation, one of my stubbed components has dependencies which are not accounted for in the context of the test. The test crashes... but it shouldn't matter because the component should be stubbed out.

**Comment by lmiller1990** [comment]
What do you mean by "has dependencies"? As in it imports other modules? The stubbing VTU offers is just before the component is rendered; it will still get imported (as well as all its dependencies).

Assuming my understanding of your question is correct, if you want to completely and totally stub out a component (and the modules it imports) you could use `jest.mock` potentially.

**Comment by robokozo** [comment]
Yeah I think you understood correctly. Your idea to use jest.mock sounds really interesting! I'll give it a go!

**Comment by lmiller1990** [comment]
Cool, here is a basic example: https://lmiller1990.github.io/vue-testing-handbook/vue-router.html#workaround-for-large-render-trees-using-mount

We should still fix the stubbing bug for <script setup>.

**Comment by freakzlike** [comment]
You can define your own way how stubs are created by a plugin (`plugins.createStubs`). https://test-utils.vuejs.org/guide/extending-vtu/plugins.html#stubs-plugin
